### PR TITLE
F# Tooling RFC FS-1034 - FSharp.Core delivery via nuget packages

### PR DESCRIPTION
**Pull request by dsyme**

### CODE REVIEW

**Comment by dsyme** [comment]
cc @enricosada @kurtschelfthout @cartermp @forki  @KevinRansom

File: RFCs/FS-1034-fsharp-core-package.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+there was a bit of a mess (made considerably worse by the separation of sigdata/optdata files, [now addressed](https://github.com/Microsoft/visualfsharp/pull/2884))
+
+The FSSF Core Engineering Group created a nuget package called [FSharp.Core](https://www.nuget.org/packages/FSharp.Core), partly to avoid
+a proliferation of "homebrew" packges appearing at that time. This packge has now grown to be a "one-stop-shop"
+package for FSharp.Core for all different target platforms. A single, unified FSharp.Core nuget packge has advantages:
+* Simplicity: Users don't have to think at all - they just reference the package and that's that
+* Multi-targeting: One packge supports building against multiple targets
+
+As of 2017, this pacakge is in good shape for the main uses envisaged by the FSSF Core Engineering Group. It has prevented
+F# users posting new, random packagings of FSharp.Core, and become a tructed part of the F# library ecosystem.  Frequently,
```
Comment (forki) [inline RFCs/FS-1034-fsharp-core-package.md]: trusted?

File: RFCs/FS-1034-fsharp-core-package.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+nuget tooling in IDE environments.
+
+As of 2017, Microsoft have a set of [time-critical objectives](https://github.com/Microsoft/visualfsharp/issues/3069) to deliver quality core tooling for F# as
+a default part of the dotnet SDK, see [this RFC](https://github.com/fsharp/fslang-design/blob/master/RFCs/FS-1032-fsharp-in-dotnet-sdk.md)
+for discussion and pros/cons.
+
+Up to this point, Microsoft have found it difficult to commit to a dependency on the community-provided
+FSharp.Core package in the dotnet SDK tooling for a number of reasons
+* The package is not signed
+* THe package is now relatively large (68MB on disk unzipped) due to the large number of variations, and may get bigger
```
Comment (forki) [inline RFCs/FS-1034-fsharp-core-package.md]: The

File: RFCs/FS-1034-fsharp-core-package.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+For the purposes of this RFC we will assume
+* Portable profiles will eventually disappear in favour of .NET Standard.  (People building PCL DLLs will still be able to reference an earlier FSharp.Core package)
+* Xamarin profiles will eventually iterate towards .NET Standard, at least for the purposes of FSharp.Core. (People building Xamarin apps will still be able to reference an earlier FSharp.Core package)
+* The runtime dependencies in the F# library ecosystem are based around DLL identity (not package identity).
+* The compile-time dependencies in the F# library ecosystem are based on package identity.  
+
+FSharp.Core is a "root dependency" in the F# library ecosystem, both as a DLL with a version and strong name dependency,
+and, increasingly, as a package.  Currently, F# libraries tend to assume either
+* The Profile 259 version of FSHarp.Core, or
```
Comment (forki) [inline RFCs/FS-1034-fsharp-core-package.md]: FSharp.Core

File: RFCs/FS-1034-fsharp-core-package.md [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+* The fatter .NET Framework 4.x version of FSharp.Core
+as their "root dependency".  Following the general principle that **libraries should have a "most portable" root dependency,
+it makes sense to 
+
+It is **very important** to note that nuget packages can easily be progressed from a current state to a different state
```
Comment (forki) [inline RFCs/FS-1034-fsharp-core-package.md]: NuGet

**Pull request merged**